kernel/aster-nix/src/net/socket/unix/stream/socket.rs

```diff
@@ -220,36 +225,28 @@ impl Socket for UnixStreamSocket {
             }
         };
 
-        let addr = init.addr().ok_or(Error::with_message(
-            Errno::EINVAL,
-            "the socket is not bound",
-        ))?;
-
-        let listener = Listener::new(addr.clone(), backlog, init.is_nonblocking())?;
-        *self.0.write() = State::Listen(Arc::new(listener));
+        let listener = Listener::new(addr, backlog)?;
+        *self.state.write() = State::Listen(listener);
         Ok(())
     }
 
     fn accept(&self) -> Result<(Arc<dyn FileLike>, SocketAddr)> {
-        let listen = match &*self.0.read() {
-            State::Listen(listen) => listen.clone(),
-            _ => return_errno_with_message!(Errno::EINVAL, "the socket is not listening"),
-        };
-
-        listen.accept()
+        if self.is_nonblocking() {
+            self.try_accept()
+        } else {
+            self.wait_events(IoEvents::OUT, || self.try_accept())
```

Comment:
There is another typo: `IoEvents::OUT` -> `IoEvents::IN`. My fault, I'm so careless. (Interestingly, the CI will pass if _both_ typos are present, but the CI will fail if the first typo is fixed).